File: src/types/activity.ts
export interface ActivityItem {
  id: string;
  title: string;
  description: string;
  status: 'pending' | 'in-progress' | 'completed';
  priority: 'low' | 'medium' | 'high' | 'urgent';
  assignee?: string;
  source: string;
  createdAt: string;
  updatedAt: string;
}

File: backend/app/schemas/activity.py
"""Activity board schemas — match frontend ActivityItem."""
from pydantic import BaseModel


class ActivityItemOut(BaseModel):
    id: str
    title: str
    description: str
    status: str  # pending | in-progress | completed
    priority: str
    assignee: str | None = None
    source: str
    createdAt: str  # frontend camelCase
    updatedAt: str
